website/addons/box/utils.py

```diff
@@ -78,6 +86,40 @@ def log(self, action, extra=None, save=False):
             self.node.save()
 
 
+def refresh_oauth_key(external_account, force=False):
```

Comment:
Fair enough. If you want to do that generalizing, let me know. Otherwise, if you could submit an Issue referencing it I'd appreciate it.